app/models/user_data.rb

```diff
@@ -24,4 +23,14 @@ def adapter
       SessionDataAdapter.new(session)
     end
   end
+
+  def create_user_keys
+    if session[:user_token].present?
```

Comment:
```suggestion
    if session[:user_token].present? && session[:user_id].blank?
```

Reply:
hmm on second thoughts no.

Reply:
i think it needs both checks coz both the user_token and user_id _could_ exist if the user has been there before

Reply:
I see. That's why I did the line below `session[:user_id] ||= session[:session_id]` **it is the same effect**. But more implicit. Do you prefer the more explicit version then?